src/context/directory/handlers/clientGrants.ts

```diff
@@ -39,15 +40,29 @@ async function dump(context: DirectoryContext): Promise<void> {
   const grantsFolder = path.join(context.filePath, constants.CLIENTS_GRANTS_DIRECTORY);
   fs.ensureDirSync(grantsFolder);
 
+  const allResourceServers = await context.mgmtClient.resourceServers.getAll({
+    paginate: true,
+  });
+
   // Convert client_id to the client name for readability
-  clientGrants.forEach((grant: Asset) => {
+  clientGrants.forEach((grant: ClientGrant) => {
     const dumpGrant = { ...grant };
 
     if (context.assets.clientsOrig) {
       dumpGrant.client_id = convertClientIdToName(dumpGrant.client_id, context.assets.clientsOrig);
     }
 
-    const name = sanitize(`${dumpGrant.client_id}`);
+    const apiName = (() => {
+      const associatedAPI = allResourceServers.find((resourceServer) => {
+        return resourceServer.identifier === grant.audience;
+      });
+
+      if (associatedAPI === undefined) return grant.audience;
+
+      return associatedAPI.name;
+    })();
+
+    const name = sanitize(`${dumpGrant.client_id}-${apiName}`);
```

Comment:
This is the bulk of the PR. Matching a client grant audience to a resource server and then using the resource server's human-friendly name.